api/src/user/controllers/role.controller.ts

```diff
@@ -142,12 +147,24 @@ export class RoleController extends BaseController<Role, RoleStub> {
   @CsrfCheck(true)
   @Delete(':id')
   @HttpCode(204)
-  async deleteOne(@Param('id') id: string) {
-    const result = await this.roleService.deleteOne(id);
-    if (result.deletedCount === 0) {
-      this.logger.warn(`Unable to delete Role by id ${id}`);
-      throw new NotFoundException(`Role with ID ${id} not found`);
+  async deleteOne(@Param('id') id: string, @Session() session: ExpressSession) {
+    const roles = (
+      await this.userService.findOneAndPopulate(session.passport?.user?.id, [
+        'roles',
+      ])
+    ).roles.map((role) => role.id);
```

Comment:
Let's break this into multiple instructions
`const currentUser = await ...`
`if (!user) { ...`
`const roles = user.roles.map(....`